proxy/src/main/java/com/wavefront/agent/GraphiteStringHandler.java

```diff
@@ -41,13 +55,28 @@ public GraphiteStringHandler(final ForceQueueEnabledAgentAPI agentAPI,
                                final long millisecondsPerBatch,
                                final int pointsPerBatch,
                                final int blockedPointsPerBatch,
-                               final Formatter formatter) {
-    this.pointHandler = new PointHandler(agentAPI, daemonId, port, logLevel, validationLevel, millisecondsPerBatch,
-        pointsPerBatch, blockedPointsPerBatch);
+                               final Formatter formatter,
+                               @Nullable final String pointLineWhiteListRegex,
+                               @Nullable final String pointLineBlackListRegex) {
+    this.pointHandler = new PointHandler(agentAPI, daemonId, port, logLevel, validationLevel,
+        millisecondsPerBatch, pointsPerBatch, blockedPointsPerBatch);
 
     this.prefix = prefix;
     this.blockedPointsPerBatch = blockedPointsPerBatch;
     this.formatter = formatter;
+
+    this.pointLineWhiteList = StringUtils.isBlank(pointLineWhiteListRegex) ?
+        null : Pattern.compile(pointLineWhiteListRegex);
+
+    this.pointLineBlackList = StringUtils.isBlank(pointLineBlackListRegex) ?
+        null : Pattern.compile(pointLineBlackListRegex);
+
+    this.regexTimer = Metrics.newTimer(
+        new MetricName("validationRegex." + String.valueOf(port), "", "microsecs"),
```

Comment:
Yeah, the timer is efficient but not free. If they have a problem, I think we can just ask them to give us the regex and we'd know (probably right away) whether it's a problem or not. Tracking every call seems a bit heavy.